src/Activity.js

```diff
@@ -0,0 +1,107 @@
+class Activity {
+  constructor(activityData) {
+    this.activityData = activityData;
+  }
+
+  findAllUserActivities(id) {
+    const activityPerUser = this.activityData.filter(activity => {
+      return activity.userID === id;
+    });
+    return activityPerUser;
+  }
+
+  findUserActivityByDate(id, date) {
+    const activityPerUser = this.findAllUserActivities(id);
+    const activityPerDate = activityPerUser.find(activity => {
+      return activity.date === date;
+    });
+    return activityPerDate;
+  }
+
+  findAllUserActivitiesByDate(date) {
+    return this.activityData.filter(activity => {
+      return activity.date === date;
+    });
+  }
+
+  getMilesWalkedForDay(id, date, stride) {
+    return Number(((this.findUserActivityByDate(id, date).numSteps * stride) / 5280).toFixed(1));
+  }
+
+  getMinutesActive(id, date) {
+    return this.findUserActivityByDate(id, date).minutesActive;
+  }
+
+  getStepGoalFeedback(id, date, dailyStepGoal) {
+    if (this.findUserActivityByDate(id, date).numSteps >= dailyStepGoal) {
+      return true;
+    }
+    if(this.findUserActivityByDate(id, date).numSteps < dailyStepGoal) {
+      return false;
+    }
```

Comment:
I think this should probably be refactored unless you had some issue that made it necessary. It seems like you could store the found user in a variable rather than searching for it twice.

Reply:
Great idea! I just refactored to store the found user like you suggested.